### ISSUE TITLE
Using the new multi encoding for mp4box to create a non multiplexed mpd.

### ISSUE THREAD

**Issue by nukebo212** [gpac/gpac#1603]
I installed gpac on my ubuntu system via performing: `sudo apt-get install gpac`  if I need to do anything else, p
I tried performing this as I came from this post: https://stackoverflow.com/questions/64057660/create-different-qaulities-for-my-manifest-with-only-mp4box

`MP4Box -dash 2000 -profile dashavc264:live test.mp4#video:@ffsws#video:osize=160x120@enc:c=avc:fintra=2:b=100k:#Representation=1@@ffsws:osize=320x240@enc:c=avc:fintra=2:b=200k:#Representation=2  test.mp4#audio -out test.mpd`

However this ends up creating a manifest file like this:

```
<?xml version="1.0"?>
<!-- MPD file Generated with GPAC version 0.5.2-DEV-revVersion: 0.5.2-426-gc5ad4e4+dfsg5-3ubuntu0.1  at 2020-10-03T04:16:26.859Z-->
<MPD xmlns="urn:mpeg:dash:schema:mpd:2011" minBufferTime="PT1.500S" type="static" mediaPresentationDuration="PT0H0M6.805S" maxSegmentDuration="PT0H0M1.984S" profiles="urn:mpeg:dash:profile:isoff-live:2011,http://dashif.org/guidelines/dash264">
 <ProgramInformation moreInformationURL="http://gpac.sourceforge.net">
  <Title>videouploads/5f77fb1ace504302517361/5f77fb1ace504302517361.mpd generated by GPAC</Title>
 </ProgramInformation>

 <Period duration="PT0H0M6.805S">
  <AdaptationSet segmentAlignment="true" maxWidth="1920" maxHeight="1080" maxFrameRate="25" par="16:9" lang="eng">
   <Representation id="1" mimeType="video/mp4" codecs="avc1.640033" width="1920" height="1080" frameRate="25" sar="1:1" startWithSAP="1" bandwidth="1827174">
    <SegmentTemplate timescale="25000" media="phpurgzST_dash_track2_$Number$.m4s" startNumber="1" duration="33000" initialization="phpurgzST_dash_track2_init.mp4"/>
   </Representation>
  </AdaptationSet>
  <AdaptationSet segmentAlignment="true" lang="eng">
   <Representation id="2" mimeType="audio/mp4" codecs="mp4a.40.2" audioSamplingRate="48000" startWithSAP="1" bandwidth="198124">
    <AudioChannelConfiguration schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2"/>
    <SegmentTemplate timescale="48000" media="phpurgzST_dash_track1_$Number$.m4s" startNumber="1" duration="95232" initialization="phpurgzST_dash_track1_init.mp4"/>
   </Representation>
  </AdaptationSet>
 </Period>
</MPD>
```

Basically it only creates 2 representations, one for video and one for audio. Not only that but the video representation has the resolution of 1920 by 1080, which is given from the source input file. I feel that this was my fault because I took the example code from the stackoverflow post above and tried specifiying the audio and video, however I don't know what to do.

Apolgies if I don't provide enough information, and I will take a while to respond to comments

Detailed guidelines: http://gpac.io/2013/07/16/how-to-file-a-bug-properly/

**Comment by jeanlf** [comment]
> 

that sys it all, you're on gpac 0.5.2 ... You must install the latest binaries from our nightly build page or compile yourself for this feature

**Comment by rbouqueau** [comment]
Current version is v1.1.0 so your package manager needs an update :)

Info about builds and building GPAC at https://github.com/gpac/gpac/wiki/Build-Introduction.

**Comment by nukebo212** [comment]
I've done that by downloading and then installing: `sudo dpkg -i gpac_1.1.0-DEV-rev57-g1759ca73-master_amd64.deb` and now I get this message everytime I try using an MP4Box command:
```
[0mint(1) array(3) { [0]=> string(33) "[31mCouldn't find HOME directory" [1]=> string(68) "[0m[31mFatal error: failed to initialize GPAC global configuration" [2]=> string(4) "[0m" }
```
Is this an issue due to permissions or how I installed gpac?

**Comment by jeanlf** [comment]
ha that makes more sense! We indeed had a bug initializing the config file, even for "0" profile (temporary one), when the execution environment had no $HOME defined. This is now fixed, thanks for the report.

**Comment by nukebo212** [comment]
Okay thanks, I reinstalled and it now has an output and I'm not getting that error message. If you don't mind I have still have one more question:

How can I add audio? When I used the command below, it creates two video representations, one in 120p and and another in 1080p. However, there isn't any audio.
```
MP4Box -fgraph -dash 2000 -profile dashavc264:live test.mp4#video:@ffsws:osize=160x120@enc:c=avc:fintra=2:b=100k:#Representation=1@@ffsws:osize=1920x1080@enc:c=avc:fintra=2:b=200k:#Representation=2  test.mp4#audio -out test.mpd
```

**Comment by nukebo212** [comment]
Also if I try any other dash command like:
```
MP4Box -dash 2000 -profile dashavc264:live -bs-switching multi -url-template test.mp4#trackID=1:id=vid0:role=vid0  test.mp4#trackID=2:id=vid1:role=vid1  test.mp4#trackID=3:id=aud0:role=aud0 -out test.mpd
```

I'll get an error like:
```
[0mError DASHing file: Internal Service Errorint(1) array(15) { [0]=> string(65) "[32m[Dasher] No template assigned, using $File$_dash$FS$$Number$" [1]=> string(132) "[0m[31m[Filter] dasher not responding properly: 100000 consecutive process with no packet discarded or sent, but 2 packets pending" [2]=> string(65) " discarding all inputs and notifying end of stream on all outputs"
```

I know that these commands worked when I was using version 0.5.2 so is it that there is a different way of dashing in the newer version?

**Comment by jeanlf** [comment]
No the cmd line is the same, this is likely source-related. Could you upload a test file?

**Comment by nukebo212** [comment]
What do you mean by that?

**Comment by nukebo212** [comment]
Oh you are right. I tried a different mp4, and it worked just fine, sorry about that. I'm still confused about how to create a manifest with audio and video with a command like:
 `MP4Box -fgraph -dash 2000 -profile dashavc264:live test.mp4#video:@ffsws:osize=160x120@enc:c=avc:fintra=2:b=100k:#Representation=1@@ffsws:osize=1920x1080@enc:c=avc:fintra=2:b=200k:#Representation=2  -out test.mpd` 
because if I include something like `test.mp4#audio` before the output there is no audio and I believe this is just due to my misunderstanding to what I'm supposed to add to such a command like this. If you also have any documentation for this specific.

**Comment by rbouqueau** [comment]
I confirm that adding a test.mp4#audio when transcoding the video has no effect.

**Comment by rbouqueau** [comment]
NB: we don't display the progress either while transcoding (may take quite some time without notifying the user).

**Comment by jeanlf** [comment]
indeed, we were not assigning inputs without filter chain as sources for the dasher, leaving them out. Now fixed and [test added](https://github.com/gpac/testsuite/blob/filters/scripts/dash-encode.sh), thanks for the report
